packages/dappkit/src/index.ts

```diff
@@ -72,41 +125,170 @@ async function getGasCurrencyContract(
 }
 
 export interface TxParams<T> {
-  txId: string
   tx: TransactionObject<T>
   from: string
   to: string
   gasCurrency: GasCurrency
+  estimatedGas?: number
+  value?: string
 }
 
 export async function requestTxSig<T>(
   web3: Web3,
   txParams: TxParams<T>[],
   meta: DappKitRequestMeta
 ) {
+  // TODO: For multi-tx payloads, we for now just assume the same from address for all txs. We should apply a better heuristic
+  const baseNonce = await web3.eth.getTransactionCount(txParams[0].from)
   const txs: TxToSignParam[] = await Promise.all(
-    txParams.map(async (txParam) => {
+    txParams.map(async (txParam, index) => {
       const gasCurrencyContract = await getGasCurrencyContract(web3, txParam.gasCurrency)
+      const value = txParam.value === undefined ? '0' : txParam.value
+
       const estimatedTxParams = {
         gasCurrency: gasCurrencyContract.options.address,
+        from: txParam.from,
+        value,
       }
-      // @ts-ignore
-      const estimatedGas = await txParam.tx.estimateGas(estimatedTxParams)
+      const estimatedGas =
+        txParam.estimatedGas === undefined
+          ? //
+            // @ts-ignore
+            await txParam.tx.estimateGas(estimatedTxParams)
+          : txParam.estimatedGas
 
-      const nonce = await web3.eth.getTransactionCount(txParam.from)
       return {
         txData: txParam.tx.encodeABI(),
         estimatedGas,
-        nonce,
+        nonce: baseNonce + index,
         gasCurrencyAddress: gasCurrencyContract._address,
+        value,
         ...txParam,
       }
     })
   )
-
-  // const url = Linking.makeUrl(returnPath)
-
   const request = SignTxRequest(txs, meta)
 
   Linking.openURL(serializeDappKitRequestDeeplink(request))
 }
+
+function isValidPhoneNumber(phoneNumber: PhoneNumber): E164Number | undefined {
+  if (phoneNumber.number === undefined) {
+    return undefined
+  }
+  const parsedPhoneNumber = parsePhoneNumberFromString(phoneNumber.number)
+
+  if (parsedPhoneNumber === undefined) {
+    return undefined
+  }
+
+  if (!parsedPhoneNumber.isValid()) {
+    return undefined
+  }
+
+  return parsedPhoneNumber.number
+}
+
+export interface PhoneNumberMappingEntry {
+  address: string
+  phoneNumber: string
+  id: string
+  attestationStat: {
+    total: number
+    completed: number
+  }
+}
+
+function createPhoneNumberToContactMapping(contacts: Contact[]) {
+  // @ts-ignore
+  const phoneNumberObjects: [{ e164Number: E164Number; id: string }] = contacts.flatMap(
```

Comment:
pretty sure `flatMap()` is rather modern. Check that supported version of node has it.

Probably doesn't fail here because of the `tsconfig` lib setting.

Reply:
in node 11: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap

Reply:
Gotcha, will revert to using lodash's flatMap